ckan/lib/dictization/model_save.py

```diff
@@ -30,6 +32,10 @@ def resource_dict_save(res_dict, context):
     table = class_mapper(model.Resource).mapped_table
     fields = [field.name for field in table.c]
 
+    # Strip the full url for resources of type 'upload'
+    if res_dict.get('url') and res_dict.get('url_type') == u'upload':
+        res_dict['url'] = res_dict['url'].rsplit('/')[-1]
+
```

Comment:
can we do the same when reading resources? That way sites changing their site_url will work even if they uploaded files before the fix.

Reply:
Doesn't look like it. I mean actually `res_dict['url'] = res_dict['url'].rsplit('/')[-1]` like above or maybe just removing the `for_edit` check in https://github.com/ckan/ckan/blob/469d068a01a33960f263142ea671aefed031547b/ckan/lib/dictization/model_dictize.py#L110

A small change like this should mean existing sites can change their SITE_URL, then run a solr reindex to update all their download urls without having to touch the db.